pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -111,7 +112,7 @@ func (c *ClusterMemberController) reconcileMembers(ctx context.Context, recorder
 	}
 
 	recorder.Eventf("FoundPodToScale", "found pod to add to etcd membership: %v", podToAdd.Name)
-	err = c.etcdClient.MemberAdd(ctx, fmt.Sprintf("https://%s:2380", etcdHost))
+	err = c.etcdClient.MemberAdd(ctx, fmt.Sprintf("https://%s", net.JoinHostPort(etcdHost, "2380")))
```

Comment:
`JoinHostPort` does what `GetEscapedPreferredInternalIPAddressForNodeName` does and since we index machines by IPs we should use `GetPreferredInternalIPAddressForNodeName` down here